cvat/apps/engine/task.py

```diff
@@ -183,25 +193,41 @@ def _validate_data(counter, meta_info_file=None):
 
     return counter, task_modes[0]
 
+
 def _download_data(urls, upload_dir):
     job = rq.get_current_job()
     local_files = {}
     for url in urls:
         name = os.path.basename(urlrequest.url2pathname(urlparse.urlparse(url).path))
-        if name in local_files:
-            raise Exception("filename collision: {}".format(name))
+        req = urlrequest.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
         slogger.glob.info("Downloading: {}".format(url))
+
         job.meta['status'] = '{} is being downloaded..'.format(url)
         job.save_meta()
 
-        req = urlrequest.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
         try:
-            with urlrequest.urlopen(req) as fp, open(os.path.join(upload_dir, name), 'wb') as tfp:
-                while True:
-                    block = fp.read(8192)
-                    if not block:
-                        break
-                    tfp.write(block)
+            with urlrequest.urlopen(req) as fp:
+                # check if file extension is present
+                if not os.path.splitext(name)[-1]:
```

Comment:
What if there is a dot inside the name? Should we try to guess the file type based on extension and if it doesn't work, use "from content type" stratage?